File: 1476-count-negative-numbers-in-a-sorted-matrix/count-negative-numbers-in-a-sorted-matrix.py
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        def BinarySearch(arr):
            n = len(arr)
            count = 0

            first = 0
            last = n - 1

            while first <= last:
                mid = (first + last) // 2

                if arr[mid] < 0:
                    last = mid - 1
                else:
                    first = mid + 1

            return first

        count = 0

        for lst in grid:
            count += len(lst) - BinarySearch(lst)

        return count

File: 1476-count-negative-numbers-in-a-sorted-matrix/count-negative-numbers-in-a-sorted-matrix.cs
public class Solution {
    public int CountNegatives(int[][] grid) {
        int m = grid.Length;
        int n = grid[0].Length;
        int count = 0;

        for (int i = 0; i < m; i++) {
            while (n > 0) {
                if (grid[i][n - 1] >= 0) {
                    break;
                }

                count += (m - i);
                n--;
            }
        }
        
        return count;
    }
}